subprojects/gradle/build-metrics/src/gradleTest/kotlin/com/avito/android/plugin/build_metrics/cache/HttpBuildCacheTestFixture.kt

```diff
@@ -75,6 +75,12 @@ internal abstract class HttpBuildCacheTestFixture {
     protected fun build(vararg args: String) =
         BuildMetricsRunner(projectDir)
             .build(
-                args.toList().plus("--build-cache")
+                args.toList().plus(listOf(
+                    "--build-cache",
```

Comment:
How do you discover this behavior?

Reply:
The bug - in debug logs. 
This is how it looks (simplified version):
```
SUCCESS

Build operation 'Apply plugin com.avito.android.build-metrics to root project
Build operation 'Snapshot task inputs for :cacheMissTask' started
Build operation 'Load entry bf51a5c9b343fe7f82bc40810a45704d from remote build cache' started
Response for PUT http://localhost:40517/bf51a5c9b343fe7f82bc40810a45704d: HTTP/1.1 500 Server Error


FAIL

Build cache key for Kotlin DSL script compilation (Project/TopLevel/stage1) is 094affd156fcef5b820c8693ad5d1f81
PUT http://localhost:37691/094affd156fcef5b820c8693ad5d1f81: HTTP/1.1 500 Server Error
Build operation 'Apply plugin com.avito.android.build-metrics to root project
Build operation 'Snapshot task inputs for :cacheMissTask' started
```

`org.gradle.unsafe.build-cache.remote-continue-on-error` - accidentally noticed it in sources while troubleshooting. 
I tried to understand what is going on. Mostly lucky that it exists already.

Unfortunately, we have no events at the moment when the remote cache is disabling, only at the end of the build: https://github.com/gradle/gradle/commit/5a343aafa0afd169b292717ee452949e3f0023b4#diff-ffe742e008b01fad18fbdbb8d9fbc3bb8b1c369a5a4a665d804031229e9ad8d5